app/src/main/java/dev/arkbuilders/rate/presentation/pairalert/PairAlertViewModel.kt

```diff
@@ -59,28 +60,45 @@ class PairAlertViewModel(
         analyticsManager.trackScreen("PairAlertScreen")
 
         intent {
-            if (isRatesAvailable().not())
+            if (currencyRepo.isRatesAvailable().not()) {
+                reduce {
+                    state.copy(noInternet = true)
+                }
                 return@intent
+            }
+
+            init()
+        }
+    }
+
+    private fun init() = intent {
+        AppSharedFlow.ShowAddedSnackbarQuick.flow.onEach { visuals ->
+            postSideEffect(PairAlertEffect.ShowSnackbarAdded(visuals))
+        }.launchIn(viewModelScope)
 
-            AppSharedFlow.ShowAddedSnackbarQuick.flow.onEach { visuals ->
-                postSideEffect(PairAlertEffect.ShowSnackbarAdded(visuals))
-            }.launchIn(viewModelScope)
-
-            pairAlertRepo.getAllFlow().onEach { all ->
-                val pages = all.reversed().groupBy { it.group }
-                    .map { (group, pairAlertList) ->
-                        val oneTimeTriggered =
-                            pairAlertList.filter { it.triggered() && it.oneTimeNotRecurrent && !it.enabled }
-                        val created = pairAlertList - oneTimeTriggered
-
-                        PairAlertScreenPage(group, created, oneTimeTriggered)
-                    }
-                intent {
-                    reduce {
-                        state.copy(pages = pages, initialized = true)
-                    }
+        pairAlertRepo.getAllFlow().onEach { all ->
+            val pages = all.reversed().groupBy { it.group }
+                .map { (group, pairAlertList) ->
+                    val oneTimeTriggered =
+                        pairAlertList.filter { it.triggered() && it.oneTimeNotRecurrent && !it.enabled }
+                    val created = pairAlertList - oneTimeTriggered
```

Comment:
I've got this suggestion. Do you think it will be improved if we follow?
<img width="650" alt="Screenshot 2024-08-03 at 19 34 25" src="https://github.com/user-attachments/assets/e9ab5089-999a-4fdc-895e-42645c2e8ccb">